lib/Dialect/Arith/Conversions/ArithToCGGI/ArithToCGGI.cpp

```diff
@@ -248,14 +332,24 @@ struct ArithToCGGI : public impl::ArithToCGGIBase<ArithToCGGI> {
     });
 
     target.addDynamicallyLegalOp<memref::StoreOp>([&](Operation *op) {
-      if (auto *defOp = cast<memref::StoreOp>(op).getValue().getDefiningOp()) {
-        if (isa<mlir::arith::ConstantOp>(defOp)) {
+      if (typeConverter.isLegal(op->getOperandTypes()) &&
+          typeConverter.isLegal(op->getResultTypes())) {
+        return true;
+      }
+
+      if (auto lweAttr =
+              op->getAttrOfType<mlir::StringAttr>("lwe_annotation")) {
+        return false;
+      }
+
+      if (auto *defOp =
+              dyn_cast<memref::StoreOp>(op).getValue().getDefiningOp()) {
```

Comment:
```suggestion
              cast<memref::StoreOp>(op).getValue().getDefiningOp()) {
```

Since the operation is guaranteed to be a StoreOp, you can use `cast` instead of `dyn_cast`